addons/purchase/tests/test_purchase_order.py

```diff
@@ -83,3 +84,82 @@ def test_00_purchase_order_flow(self):
         })
         self.invoice.purchase_order_change()
         self.assertEqual(self.po.order_line.mapped('qty_invoiced'), [5.0, 5.0], 'Purchase: all products should be invoiced"')
+
+    def test_01_vendor_price(self):
+        #Case1: We find a vendor for the product variant and take the variant vendor price.
+        #Add the vendors in product variants.
+        supplier_info_1 = {
+            'product_seller_ids': [
+                (0, 0, {
+                    'name': self.partner_id.id,
+                    'min_qty': 5.0,
+                    'price': 500.0,
+                })
+            ]
+        }
+        self.product_id_1.write(supplier_info_1)
+
+        #Lets create a purchase order to test the vendor price.
+        po_vals_1 = {
+            'partner_id': self.partner_id.id,
+        }
+        self.po1 = self.PurchaseOrder.create(po_vals_1)
+
+        po_line_dict_1 = {
+            'name': self.product_id_1.name,
+            'product_id': self.product_id_1.id,
+            'order_id': self.po1.id,
+            'product_uom': self.product_id_1.uom_po_id.id,
+            'date_planned': datetime.today().strftime(DEFAULT_SERVER_DATETIME_FORMAT),
+        }
+
+        #Call the 'onchange_product_id' to get the price from vendor.
+        po_line_1 = self.PurchaseOrderLine.new(po_line_dict_1)
```

Comment:
Here i want to create a record based on data provided by on-change method to test the feature.